client/client.go

```diff
@@ -403,6 +404,13 @@ type (
 	}
 )
 
+type MetricsHandler = metrics.Handler
+type MetricsCounter = metrics.Counter
+type MetricsGauge = metrics.Gauge
+type MetricsTimer = metrics.Timer
```

Comment:
I had understood that we do not expect client code to use our internal instrument types. I.e. the  expectation is that they instrument and export their own metrics with whatever tools they prefer. Do I have that right?

Reply:
I believe that is correct. This abstraction is very specific to our uses. This is not a general-purpose metrics abstraction. We definitely expect their metric use to leverage whatever they back the abstraction with (e.g. use the Tally scope directly), not the abstraction itself.

Technically they _could_ use our abstraction if they're uses were as limited as ours (e.g. no histograms). We do expose `activity.GetMetricsHandler` and `workflow.GetMetricsHandler` for this purpose which does have replay protection and some tags built-in, but unless they really need those things, I'd discourage it. Assuming the team is OK with this PR, I'll make all of this clear in the docs.